frontends/web/src/routes/account/summary/summarybalance.tsx

```diff
@@ -55,73 +77,79 @@ export function SummaryBalance ({
-          <tr>
-            <th>{t('accountSummary.name')}</th>
-            <th>{t('accountSummary.balance')}</th>
-            <th>{t('accountSummary.fiatBalance')}</th>
-          </tr>
-        </thead>
-        <tbody>
-          { accounts.length > 0 ? (
-            coins.map(coinCode => {
-              const balanceRows = accountsPerCoin[coinCode].map(account =>
-                <BalanceRow
-                  key={account.code}
-                  code={account.code}
-                  name={account.name}
-                  coinCode={account.coinCode}
-                  balance={balances && balances[account.code]}
-                />
-              );
-              if (balanceRows?.length > 1) {
-                const account = accountsPerCoin[coinCode][0];
-                balanceRows.push(
-                  <SubTotalRow
-                    key={account.coinCode}
+    <div>
+      <div className={style.accountName}>
+        <p>{keystoreName} {keystoreDisambiguatorName && `(${keystoreDisambiguatorName})`}</p>
+        {connected ?
+          <Badge
+            icon={props => <USBSuccess {...props} />}
+            type="success"
+          >
+            {t('device.keystoreConnected')}
+          </Badge> :
+          null
+        }
+      </div>
+      <div className={style.balanceTable}>
+        <table className={style.table}>
+          <colgroup>
+            <col width="33%" />
+            <col width="33%" />
+            <col width="*" />
+          </colgroup>
+          <thead>
+            <tr>
+              <th>{t('accountSummary.name')}</th>
+              <th>{t('accountSummary.balance')}</th>
+              <th>{t('accountSummary.fiatBalance')}</th>
+            </tr>
+          </thead>
+          <tbody>
+            { accounts.length > 0 ? (
+              coins.map(coinCode => {
+                const balanceRows = accountsPerCoin[coinCode].map(account =>
+                  <BalanceRow
+                    key={account.code}
+                    code={account.code}
+                    name={account.name}
                     coinCode={account.coinCode}
-                    coinName={account.coinName}
-                    balance={totalBalancePerCoin && totalBalancePerCoin[coinCode]}
-                  />);
-              }
-              return balanceRows;
-            })
-          ) : (
+                    balance={balances && balances[account.code]}
+                  />
+                );
+                if (balanceRows?.length > 1) {
+                  const account = accountsPerCoin[coinCode][0];
+                  balanceRows.push(
+                    <SubTotalRow
+                      key={account.coinCode}
+                      coinCode={account.coinCode}
+                      coinName={account.coinName}
+                      balance={totalBalancePerCoin && totalBalancePerCoin[coinCode]}
+                    />);
+                }
+                return balanceRows;
+              })
```

Comment:
@Beerosagos I was referring to this whole condition. Doing balanceRows.push inside the whole component looks a bit atypical, to me. I'd personally take it out into a new function, maybe like `renderBalanceRows(coins)` or smt like that.

But if it's clear to you then no big deal tbh :smile: :+1: